vars/publishToNuget.groovy

```diff
@@ -19,9 +19,10 @@
 void call(Map args = [:]) {
     lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
     checkout([$class: 'GitSCM', branches: [[name: "${args.tag}" ]], userRemoteConfigs: [[url: "${args.repository}" ]]])
+
     sh """
     dotnet build ${WORKSPACE}/${args.solutionFilePath} --configuration Release
-    find src -name OpenSearch*.dll>${WORKSPACE}/dlls.txt
+    find src/OpenSearch.*/bin/Release/*/*.dll -type f -regextype posix-extended -regex "src/([^/]+)/bin/Release/[^/]+/\\1\\.dll">${WORKSPACE}/dlls.txt
```

Comment:
Can you explain what this regex is trying to do?
Thanks.

Reply:
Get all OpenSearch dll files that needs to be signed. Looks like dlls are duplicated across the folders while building so we just want to sign want will be released. I got this from @Xtansia reason is mentioned in PR description.

Reply:
For context the "true" DLLs that needs signing and that will be packaged, for each project is:
`src/{project_name}/bin/Release/{target_framework}/{project_name}.dll`
But all the dependency DLLs also get copied into `bin/Release/{target_framework}` and there's also copies in an `obj` so can't just blindly take all DLLs. So this regex captured the `{project_name}` and uses a back reference `\1` to make sure it finds the DLL with the same name as the project.
You can see the difference in output between the previous version and this one in this screenshot:
<img width="1131" alt="Screenshot 2023-03-15 at 12 53 14 PM" src="https://user-images.githubusercontent.com/1222964/225442638-108a9a72-64cd-48ca-8f10-caa77382dbbe.png">

Reply:
Got it, thanks both for the great explanation.